ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/AbstractBlockProcessor.java

```diff
@@ -697,7 +681,21 @@ public boolean verifyVoluntaryExits(
     return true;
   }
 
+  // Catch generic errors and wrap them in a BlockProcessingException
+  private void safelyProcess(BlockProcessingAction action) throws BlockProcessingException {
+    try {
+      action.run();
+    } catch (IllegalArgumentException | IndexOutOfBoundsException e) {
+      LOG.warn(e.getMessage());
```

Comment:
Can we include the full exception here please? It's unusual to get and we already exclude stack traces when printing to the console so it should work out pretty well.  I've been finding the messages we get when blocks are invalid to be too limited so this would be a good first step to improving them.